CMakeLists.txt

```diff
@@ -4,11 +4,12 @@ project(SummerShuffle LANGUAGES CXX CUDA)
 set(CMAKE_CXX_STANDARD 17)
 include_directories(SYSTEM ${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES})
 include_directories(${CMAKE_SOURCE_DIR}/include)
+include_directories(${CMAKE_SOURCE_DIR}/vendor/nvidia/cub)
 
-set( CUDA_NVCC_FLAGS ${CUDA_NVCC_FLAGS} )
+set(CUDA_NVCC_FLAGS ${CUDA_NVCC_FLAGS} )
 
-#add_executable(main main.cu)
-#target_compile_options(main PRIVATE $<$<COMPILE_LANGUAGE:CUDA>:--expt-extended-lambda>)
+add_executable(main main.cu)
+target_compile_options(main PRIVATE $<$<COMPILE_LANGUAGE:CUDA>:--expt-extended-lambda;--std=c++14>)
```

Comment:
Ive been using c++17. Whats wrong with this way https://github.com/djns99/CUDA-Shuffle/pull/2/files#diff-af3b638bc2a3e6c650974192a53c7291L4

Reply:
Your cake flag affects the host compiler but not nvcc. The std=c++17 flag is not understood by nvcc, but I could add c++14.

I found I had to add this flag to get it to compile, I think my toolchain is a little different to yours.

Reply:
Ok that makes sense. It might be nice to use https://cmake.org/cmake/help/v3.10/prop_tgt/COMPILE_OPTIONS.html or else an environment variable so we dont have to add it in 3 places